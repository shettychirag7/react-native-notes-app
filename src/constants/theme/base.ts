import {Theme} from './type';
import {fonts} from '../fonts';

const base: Theme = {
  fontFamily: {
    titleLight: fonts.montserrat.light,
    titleRegular: fonts.montserrat.semiBold,
    titleBold: fonts.montserrat.bold,
    bodyLight: fonts.montserrat.thin,
    bodyBold: fonts.montserrat.medium,
    bodyRegular: fonts.montserrat.regular,
  },
  fontSize: {
    sm: 14,
    md: 16,
    lg: 20,
    xl: 24,
    xxl: 32,
  },
  space: {
    xs: 4,
    sm: 8,
    md: 16,
    lg: 24,
    xl: 32,
    xxl: 40,
  },
  colors: {
    primary: {
      10: '#aceef5',
      20: '#70e3f0',
      30: '#00d7ea',
      40: '#00cee6',
      50: '#00c4e1',
      60: '#00b4ce',
      70: '#009fb3',
      80: '#008b9a',
      90: '#00676c',
    },
    secondary: {
      10: '#f5f5f5',
      20: '#eeeeee',
      30: '#e0e0e0',
      40: '#bebebe',
      50: '#9f9f9f',
      60: '#767676',
      70: '#626262',
      80: '#424242',
      90: '#212121',
    },
    neutral: {
      N10: '#F7F7F7',
      N20: '#E1E1E1',
      N30: '#A8A8A8',
      N40: '#757575',
      N50: '#565656',
      N60: '#3B3B3B',
      N70: '#222222',
      N80: '#121212',
      white: '#FFFFFF',
      black: '#000000',
    },
  },
};

export default base;
