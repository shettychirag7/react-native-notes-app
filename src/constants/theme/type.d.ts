export type Theme = {
  fontFamily: {
    titleLight: string;
    titleRegular: string;
    titleBold: string;
    bodyLight: string;
    bodyBold: string;
    bodyRegular: string;
  };
  fontSize: {
    sm: number;
    md: number;
    lg: number;
    xl: number;
    xxl: number;
  };
  space: {
    xs: number;
    sm: number;
    md: number;
    lg: number;
    xl: number;
    xxl: number;
  };
  colors: {
    primary: {
      10: string;
      20: string;
      30: string;
      40: string;
      50: string;
      60: string;
      70: string;
      80: string;
      90: string;
    };
    secondary: {
      10: string;
      20: string;
      30: string;
      40: string;
      50: string;
      60: string;
      70: string;
      80: string;
      90: string;
    };
    neutral: {
      N10: string;
      N20: string;
      N30: string;
      N40: string;
      N50: string;
      N60: string;
      N70: string;
      N80: string;
      white: string;
      black: string;
    };
  };
};
