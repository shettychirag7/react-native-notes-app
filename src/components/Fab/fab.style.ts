import {StyleSheet} from 'react-native';

import {createStylePickerHook, GeneratorProps} from '../../hooks/style';

const getStyleObject = ({theme}: GeneratorProps) =>
  StyleSheet.create({
    fab: {
      opacity: 0.8,
      height: 60,
      width: 60,
      justifyContent: 'center',
      alignItems: 'center',
      backgroundColor: theme.colors.neutral.N50,
      shadowColor: theme.colors.neutral.N80,
      shadowOffset: {width: 0, height: -1},
      shadowOpacity: 0.5,
      elevation: 5,
      borderRadius: 50,
    },
  });

const styles = createStylePickerHook(getStyleObject);

export default styles;
