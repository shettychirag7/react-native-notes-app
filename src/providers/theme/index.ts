import {createContext} from 'react';
import themes from '../../constants/theme';

export const ThemeContext = createContext({
  theme: themes.base,
});

export const ThemeProvider = ThemeContext.Provider;
