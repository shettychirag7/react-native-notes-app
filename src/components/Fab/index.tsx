import React, {useContext} from 'react';
import {TouchableOpacity, TouchableOpacityProps} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import {ThemeContext} from '../../providers/theme';
import useStylePicker from './fab.style';

interface FabProps extends TouchableOpacityProps {
  icon: string;
}

const Fab = ({icon, style, ...rest}: FabProps) => {
  const styles = useStylePicker();
  const {theme} = useContext(ThemeContext);
  return (
    <TouchableOpacity style={[styles.fab, style]} {...rest}>
      <Icon name={icon} color={theme.colors.neutral.white} size={30} />
    </TouchableOpacity>
  );
};

export default Fab;
