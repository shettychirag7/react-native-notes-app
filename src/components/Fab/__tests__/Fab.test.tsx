import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';

import Fab from '..';
import {ThemeProvider} from '../../../providers/theme';
import themes from '../../../constants/theme';

describe('ListItem component test', () => {
  it('renders ListItem component', () => {
    const {toJSON} = render(
      <ThemeProvider value={{theme: themes.base}}>
        <Fab icon="home" />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('attaches onPress callback', () => {
    const mockOnPress = jest.fn();
    const {getByTestId} = render(
      <ThemeProvider value={{theme: themes.base}}>
        <Fab icon="home" testID="fab" onPress={mockOnPress} />
      </ThemeProvider>,
    );

    fireEvent.press(getByTestId('fab'));

    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });
});
