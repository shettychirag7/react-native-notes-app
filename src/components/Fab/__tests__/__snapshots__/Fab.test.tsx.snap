// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component test renders ListItem component 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#565656",
      "borderRadius": 50,
      "elevation": 5,
      "height": 60,
      "justifyContent": "center",
      "opacity": 0.8,
      "shadowColor": "#121212",
      "shadowOffset": Object {
        "height": -1,
        "width": 0,
      },
      "shadowOpacity": 0.5,
      "width": 60,
    }
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontSize": 30,
        },
        undefined,
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    î¢Š
  </Text>
</View>
`;
